content/Link Tracking with StimulusReflex - Part II.md

```diff
@@ -0,0 +1,613 @@
+              <% end %>
+            </ul>
+          </div>
+        <% end %>
+
+        <div class="form-group">
+          <%= form.label :name %>
+          <%= form.text_field :name, class: "form-control" %>
+        </div>
+
+        <div class="form-group">
+          <%= form.label :shortened_path %>
+          <%= form.text_field :shortened_path, class: "form-control" %>
+        </div>
+
+        <div class="form-group">
+          <%= form.label :original_url %>
+          <%= form.text_field :original_url, class: "form-control" %>
+        </div>
+
+        <button class="btn btn-primary">Save</button>
+      <% end %>
+    </div>
+  </div>
+</div>
+```
+
+With the needed CSS in `app/assets/stylesheets/modal.css`
+
+```css
+.live-modal-backdrop {
+    opacity: 1 !important;
+    position: fixed;
+    z-index: 1;
+    left: 0;
+    top: 0;
+    width: 100%;
+    height: 100%;
+    overflow: auto;
+    background-color: #FFF;
+    background-color: rgba(0,0,0,0.4);
+}
+
+.live-modal-content {
+    background-color: #fefefe;
+    margin: 15% auto;
+    padding: 0 20px 20px 20px;
+    border: 1px solid #888;
+    width: 80%;
+}
+
+.live-modal-close {
+    color: #aaa;
+    text-align: end;
+    font-size: 28px;
+    font-weight: bold;
+}
+
+.live-modal-close:hover,
+.live-modal-close:focus {
+    color: black;
+    text-decoration: none;
+    cursor: pointer;
+}
+
+```
+
+There's not much to be said here, the only interesting part is `<div class="live-modal" data-controller="shortened-link-form">` where we are actually using a new controller. This is just to decompose our code in more reusable units.
+
+Let's go ahead and add that Stimulus controller in `app/javascript/controllers/shortened_link_form_controller.js` with a method to close the modal
+
+```javascript
+import { Controller } from "stimulus"
+import StimulusReflex from 'stimulus_reflex'
+
+export default class extends Controller {
+  connect () {
+    StimulusReflex.register(this)
+  }
+
+  closeModal() {
+    this.stimulate('ShortenedLinkFormReflex#close_modal')
+  }
+}
+```
+
+For which we'll clearly need a new reflex in  `app/reflexes/shortened_link_form_reflex.rb` with the following code
+
+```ruby
+class ShortenedLinkFormReflex < ApplicationReflex
+  def close_modal
+    @form_action = :none
+  end
+end
+```
+
+Pretty reactive, we just need to change a variable and the `if` we added above will automatically stop rendering the modal by not sending the HTML to the frontend. Very neat in my opinion.
+
+Next up we just bind the closing action on the `X`
+
+```html
+<div class="live-modal-close" data-action="click->shortened-link-form#closeModal">&times;</div>
+```
+
+And on the backdrop
+
+```html
+<div class="live-modal-backdrop" data-action="click->shortened-link-form#closeModal">
+```
+
+But we also need to trap the click so that it doesn't close when clicking inside. So we just add a method to trap that `app/javascript/controllers/shortened_link_form_controller.js`
```

Comment:
```suggestion
But we also need to trap the click so that it doesn't close when clicking inside. So we just add this method to trap that - `app/javascript/controllers/shortened_link_form_controller.js`
```